extensions/spring/spring-boot-autoconfigure/src/main/java/org/axonframework/extension/springboot/autoconfig/CustomJacksonModuleAutoConfiguration.java

```diff
@@ -0,0 +1,147 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.extension.springboot.autoconfig;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.Module;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import org.axonframework.extension.springboot.ConverterProperties;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.AnyNestedCondition;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Autoconfiguration class dedicated to providing custom Jackson modules for the {@link ObjectMapper}.
+ * <p>
+ * This autoconfiguration is registered before the {@link ObjectMapperAutoConfiguration} and
+ * {@link AxonAutoConfiguration} to ensure custom modules are available when the {@link ObjectMapper} is
+ * constructed.
+ *
+ * @author Theo Emanuelsson
+ * @since 5.0.0
+ */
+@AutoConfiguration
+@AutoConfigureBefore({AxonAutoConfiguration.class, ObjectMapperAutoConfiguration.class})
+@ConditionalOnClass(name = "com.fasterxml.jackson.databind.ObjectMapper")
+@EnableConfigurationProperties(value = ConverterProperties.class)
+public class CustomJacksonModuleAutoConfiguration {
+
+    /**
+     * Returns a Jackson {@link Module} that provides a custom deserializer for the Spring Data {@link Page} interface.
+     * <p>
+     * This {@code Module} bean is only created when the Spring Data {@link Page} interface is on the classpath
+     * <b>and</b> whenever the user specified either the {@link ConverterProperties.ConverterType#DEFAULT} or
+     * {@link ConverterProperties.ConverterType#JACKSON} {@code ConverterType}. The module deserializes {@link Page}
+     * instances into {@link PageImpl} objects.
+     *
+     * @return A Jackson {@link Module} that deserializes {@link Page} into {@link PageImpl}.
+     * @see JacksonPageDeserializer
+     */
+    @Bean
+    @ConditionalOnClass(Page.class)
+    @Conditional(JacksonConfiguredCondition.class)
+    public Module springDataPageJacksonModule() {
```

Comment:
I think it's nicer if we group this bean creation method in the `ObjectMapperAutoConfiguration`, as it's `ObjectMapper` details that we are introducing.